src/main/java/eu/europa/ted/efx/xpath/XPathAttributeLocator.java

```diff
@@ -25,11 +28,11 @@ public class XPathAttributeLocator extends XPath20BaseListener {
 
   private int inPredicate = 0;
   private int splitPosition = -1;
-  private PathExpression path;
+  private String path;
   private String attribute;
 
-  public PathExpression getPath() {
-    return path;
+  public NodePathExpression getPath() {
+    return Expression.instantiate(path, NodePathExpression.class);
```

Comment:
I guess this returns the XPath without the attribute. This should be indicated explicitly, at least in a javadoc method comment.